### ISSUE TITLE
Front matter support

### ISSUE THREAD

**Issue by gglanzani** [dullage/flatnotes#200]
I was wondering if front matter support (via yaml, f.e.) will be available in the future.

It would allow extra metadata such as date

```
---
title: Remote Working Policy
author: Giovanni Lanzani
date: 2024.06.14
---
```

**Comment by dullage** [comment]
This isn't something that I have any plans for, but wouldn't be against it if there were sufficient demand.

**Comment by w33ble** [comment]
I just started using flatnotes (nice app btw) and was surprised it wasn't already using front matter. It's more common in flat file blogging platforms in my experience, but I had assumed going in that that's where it would keep track of tags and other metadata. I also imagine that knowing the creation date would be helpful as your notes library grows, and creation date could possibly be another sorting option too.

**Comment by deuts** [comment]
I agree with adding front matter support. So my markdown notes will also be usable across other platforms. Thanks.

**Comment by iocouto** [comment]
The use-cases for flatnotes would grow exponentially if front matter searching was supported:

- it could be used as a headless blog CMS, if able to search by information such as `author` and `published-date`.
- it could be used as a headless catalog CMS for eCommerce, if able to search by info such as `SKU`, `type` and `manufacturer`
- it could be used as a headless backend for a support ticketing system, if able to search by `ID`, `status`, `assigned`, and more.

**Comment by jhult** [comment]
For reference, [here](https://joplinapp.org/help/dev/spec/interop_with_frontmatter/) is Joplin's front matter spec.

**Comment by damufo** [comment]
I have been using silverbullet, with more than 1500 notes files, currently it is too slow for me.
I am looking for alternatives.
Flatnotes could be one.
In this case my files have names like zettelkasten so the title/filename is not that informative for me.
That is why at the beginning of the document I use yaml tags to indicate the title of the document, which is also less restrictive with the characters, for example I can put:
```
title: "Book: This is a book title"
author: "author name"
```

**Comment by deuts** [comment]
> I have been using silverbullet, with more than 1500 notes files, currently it is too slow for me.

I thought it was only me who found SB too slow.

**Comment by labrunning** [comment]
I've just discovered flatnotes and it's really scratched my self hosted notes itch. I've found the only thing I miss so far is support for front matter. I'm not using the web view much because of this, thankfully it's not a deal breaker in front matter which I mainly use for access from my phone. When at a PC I simply ssh into my server to access my notes.
